### PR TITLE
Bundle2

### LINKED ISSUE #9: Rewrite unit tests
Rewrite the unit tests in nUnit or xUnit.
This guide might be a good place to start http://approache.com/blog/from-mstest-to-nunit/

### LINKED ISSUE #16: Sensitive data option

### LINKED ISSUE #17: Message templates

### LINKED ISSUE #18: Document options

### LINKED ISSUE #19: Folder hierarchy

### PR DESCRIPTION
**Pull request by chrsadk**
This PR fixes #9, fixes #16,  fixes #17, fixes #18 and fixes #19

### CODE REVIEW

**Comment by chrsadk** [comment]
This is the last "bundle" pull request. Future pull requests will be send for each individual issue.

**Comment by janflora** [comment]
@chrsadk I've updated the message to make sure that all the related issues gets closed when we merge the pull request.

File: Src/Penneo/(Model)/DocumentOption.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Penneo
```
Comment (janflora) [inline Src/Penneo/(Model)/DocumentOption.cs]: This format is not correct.

File: Src/Penneo/(Model)/Folder.cs [hunk 2]
```diff
@@ -31,5 +40,50 @@ public bool RemoveCaseFile(CaseFile caseFile)
         {
             return UnlinkEntity(caseFile);
         }
+
+        public override string ToString()
+        {
+            return Title + "(" + Id + (ParentFolder != null ? "," + ParentFolder.Id : null) + ")";
+        }
+
+        public static int GetDept(List<Folder> list, Folder folder)
```
Comment (janflora) [inline Src/Penneo/(Model)/Folder.cs]: "Depth" is mis-spelled.

**Pull request merged**